salt/states/cmd.py

```diff
@@ -912,10 +923,11 @@ def run(name,
 
     # Wow, we passed the test, run this sucker!
     try:
-        cmd_all = __salt__['cmd.run_all'](
-            name, timeout=timeout, python_shell=True, **cmd_kwargs
+        run_cmd = 'cmd.run_all' if not root else 'cmd.run_chroot'
+        cmd_all = __salt__[run_cmd](
+            cmd=name, timeout=timeout, python_shell=True, **cmd_kwargs
         )
-    except CommandExecutionError as err:
+    except Exception as err:
```

Comment:
Should we be catching all `Exceptions` here? 🤔

Reply:
Yes, this comes from the review of the original PR:

https://github.com/saltstack/salt/pull/50302#discussion_r229833477

The logic here is that we are making a call to **any** salt module (e.g mkdir, file.managed, etc), so we can have any kind of exception, including some declared by the user (via a _module)